admin/modules/config/smilies.php

```diff
@@ -643,7 +643,7 @@
 	while($smilie = $db->fetch_array($query))
 	{
 		$smilie['image'] = str_replace("{theme}", "images", $smilie['image']);
-		if(my_strpos($smilie['image'], "p://") || substr($smilie['image'], 0, 1) == "/")
+		if(filter_var($smilie['image'], FILTER_VALIDATE_URL))
```

Comment:
`FILTER_VALIDATE_URL` doesn't allow relative paths (2nd part of the conditional), does it?